### DISCUSSION TITLE
How to upgrade releases?

### DISCUSSION THREAD

**Discussion by MakiseKurisu** [nix-community/nix-on-droid#349] [Q&A] [answered]
It must be really obvious since I can't be the only one want to do this, yet I can't find the answer...

My system was initially set up with 2023.05. I'm currently trying to convert it into flakes, but I faced hardcoded proot package missing. I tried to go back to a classic setup but the result is the same:

```
-bash-5.2$ nix-on-droid build                                                                                                
Building activation package...
error: path '/nix/store/7w09z1kw62wg7nv3q3z2p6kxf1ihk178-proot-termux-static-aarch64-unknown-linux-android-unstable-2023-11-11' does not exist and cannot be created
(use '--show-trace' to show detailed location information)
-bash-5.2$ nix-channel --list
home-manager https://github.com/nix-community/home-manager/archive/release-23.11.tar.gz
nix-on-droid https://github.com/nix-community/nix-on-droid/archive/release-23.11.tar.gz
nixpkgs https://mirrors.ustc.edu.cn/nix-channels/nixos-23.11
nixpkgs-unstable https://mirrors.ustc.edu.cn/nix-channels/nixos-unstable
-bash-5.2$
```

Do I have to clear the data and bootstrap again?

**Comment by t184256** [comment]
On April 23, 2024 4:53:04 PM GMT+02:00, "牧瀬紅莉栖" ***@***.***> wrote:
>It must be really obvious since I can't be the only one want to do this, yet I can't find the answer...
>
>My system was initially set up with 2023.05. I'm currently trying to convert it into flakes, but I faced hardcoded proot package missing. I tried to go back to a classic setup but the result is the same:
>
>```
>-bash-5.2$ nix-on-droid build                                                                                                
>Building activation package...
>error: path '/nix/store/7w09z1kw62wg7nv3q3z2p6kxf1ihk178-proot-termux-static-aarch64-unknown-linux-android-unstable-2023-11-11' does not exist and cannot be created
>(use '--show-trace' to show detailed location information)
>-bash-5.2$ nix-channel --list
>home-manager https://github.com/nix-community/home-manager/archive/release-23.11.tar.gz
>nix-on-droid https://github.com/nix-community/nix-on-droid/archive/release-23.11.tar.gz
>nixpkgs https://mirrors.ustc.edu.cn/nix-channels/nixos-23.11
>nixpkgs-unstable https://mirrors.ustc.edu.cn/nix-channels/nixos-unstable
>-bash-5.2$
>```
>
>Do I have to clear the data and bootstrap again?
>

Theoretically, the right proot should be pinned at cachix...

**Reply by MakiseKurisu** [reply, marked as answer]
Yep I can see the package [there](https://app.cachix.org/cache/nix-on-droid#pins), but I don't seem to have cachix configured in my current system.
Do I need to install cachix first and do `cachix use nix-on-droid`?

**Reply by Gerschtli** [reply]
Either that or specify the cache via cli options when running nix-on-droid switch
